src/rgw/rgw_common.h

```diff
@@ -1108,6 +1118,7 @@ struct RGWBucketInfo {
   bool has_instance_obj{false};
   RGWObjVersionTracker objv_tracker; /* we don't need to serialize this, for runtime tracking */
   RGWQuotaInfo quota;
+  req_state* s = nullptr;
```

Comment:
bucket info does not need req_state pointer

Reply:
in many places bucket_info is passed as parameter to function which requires tracing, so I thought of introducing req_state inside it so that it be able to access the stored spans in `stack_span` variable.

Reply:
please don't do that. adding this dependency is incorrect (see the general comment)

Reply:
@yuvalif I saw your comment but if try to do tracing by passing them as function parameter then there will be lots of code invloved as I have to make a overload function of every function I want to trace, and this will create a huge code.

Reply:
@yuvalif can I put my `Span` variable instead of `req_state`?

Reply:
or... we can decide which flows we really want to trace. IMO, it is important to trace the main flow end2end, but less important to trace each side branch of it.
see, for example, the way that added coroutine support was added into our code (it is very similar to what you are trying to do here).
we added an `optional_yield` parameter to each function that needed it (without overloading). and structured it in a way that it would be a no-op if no one initialized it at the beginning of the chain.

classes should have well defined meanings, and "BucketInfo" should not contain members that has nothing to do with buckets.
the one exception is req_state, only because it is already a huge collection of unrelated stuff :-(

Reply:
@yuvalif I was thinking the same now an optional parameter, but the thing is Span is unique_ptr I cannot define it like 
`func(Span span = nullptr)` so I need to make it a shared_ptr now. Currently I m making other changes you mentioned, implementing this change will take some time though.

Can I do this 
```
class Bucket_Info{
.
. 
#ifdef  
Span span = nullptr;
#end 
}
```
not only in this one but other class also like RGWRados::Object.

Reply:
one more thing if I implement like this `func(Span span){}` then it will be compile error when code will compile `WITH_JAEGER` off so in rgw_common.h I need to include `tracer.h` without `WITH_JAEGER` directive. Isnt it?

Reply:
@yuvalif I m moving forward with introducing `Span` as data member instead of `req_state`

Reply:
noted

Reply:
just to make sure we are on the same page:
* add span to req_state, and use it to trace where req_state exist
* **do not add req_state to other objects or functions**
* if you need to trace functions that don't have access to req_state, please add an `optional_span` to them, similar to how we add `optional_yield` to our functions. if these functions originated from the beast frontend, their `optional_span` will be initialized, and could be used for tracing. if the same functions are invoked from somewhere else (e.g. `radosgw-admin`, or civetweb frontend), their `optional_span` will be null, and no tracing would happen

Reply:
I m currently working removing req_state from classes instead adding `span` as data member in a seperate branch it will take some time though. and I am not using this `optional_span` because injecting span as data member will do the task for me, should I proceed with this approach?

Reply:
we agreed that injecting span as a data member to `req_state` is acceptable, and if you are good with tracing only places that have access to `req_state` then we are good.
however, for other places, please use `optional_span` and not the span as a data member to all sorts of classes.
remember that these calls (e.g. create_bucket()) may be invoked from places which are not the frontend (or even the rgw)